### ISSUE TITLE
Code Climate index out of range with unused linter since > v1.21.0

### ISSUE THREAD

**Issue by arminbuerkle** [golangci/golangci-lint#979]
Thank you for creating the issue!

- [x] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [x] Yes, I've searched similar issues on GitHub and didn't find any.
- [x] Yes, I've included all information below (version, config, etc).

Please include the following information:

<details><summary>Version of golangci-lint</summary>

```console
$ golangci-lint --version
golangci-lint has version 1.23.7 built from 96964db on 2020-02-28T12:07:44Z
```

</details>

<details><summary>Config file</summary>

```console
$ cat .golangci.yml
cat: .golangci.yml: No such file or directory
```

</details>

<details><summary>Go environment</summary>

```console
$ go version && go env
go version go1.13.8 linux/amd64
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/root/.cache/go-build"
GOENV="/root/.config/go/env"
GOEXE=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/golang/go.mod"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build155065711=/tmp/go-build -gno-record-gcc-switches"
```

</details>

<details><summary>Verbose output of running</summary>

```console
$ golangci-lint run -v --disable-all -E unused --out-format code-climate
INFO [config_reader] Config search paths: [./ /golang /] 
INFO [lintersdb] Active 1 linters: [unused]       
INFO [lintersdb] Active 1 linters: [unused]       
INFO [loader] Go packages loading at mode 575 (types_sizes|compiled_files|exports_file|name|deps|files|imports) took 213.082062ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 2.243655ms 
INFO [runner/unused/goanalysis] analyzers took 0s with no stages 
INFO [runner] Processors filtering stat (out/in): autogenerated_exclude: 4/4, diff: 4/4, max_per_file_from_linter: 4/4, filename_unadjuster: 4/4, skip_files: 4/4, identifier_marker: 4/4, exclude: 4/4, uniq_by_line: 4/4, max_from_linter: 4/4, path_shortener: 4/4, cgo: 4/4, path_prettifier: 4/4, nolint: 4/4, max_same_issues: 4/4, source_code: 4/4, skip_dirs: 4/4, exclude-rules: 4/4 
INFO [runner] processing took 1.029837ms with stages: nolint: 851.63µs, exclude: 75.418µs, identifier_marker: 35.178µs, autogenerated_exclude: 22.674µs, path_prettifier: 19.401µs, skip_dirs: 9.853µs, max_same_issues: 3.652µs, uniq_by_line: 3.158µs, max_from_linter: 1.976µs, path_shortener: 1.611µs, cgo: 1.43µs, filename_unadjuster: 1.005µs, source_code: 933ns, max_per_file_from_linter: 830ns, skip_files: 538ns, diff: 326ns, exclude-rules: 224ns 
INFO [runner] linters took 56.885007ms with stages: unused: 55.778981ms 
INFO Memory: 4 samples, avg is 71.5MB, max is 72.1MB 
INFO Execution took 276.702411ms                  
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/golangci/golangci-lint/pkg/printers.CodeClimate.Print(0x10f14a0, 0xc000185b00, 0xc0011b4000, 0x4, 0x4, 0xc0001aad80, 0xfa8660)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/printers/codeclimate.go:44 +0x52f
github.com/golangci/golangci-lint/pkg/commands.(*Executor).runAndPrint(0xc0000b2a00, 0x10f14a0, 0xc000185b00, 0xc000185a40, 0x0, 0x6, 0x0, 0x0)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/commands/run.go:372 +0x324
github.com/golangci/golangci-lint/pkg/commands.(*Executor).executeRun(0xc0000b2a00, 0xc000201b80, 0xc000185a40, 0x0, 0x6)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/commands/run.go:423 +0x17b
github.com/spf13/cobra.(*Command).execute(0xc000201b80, 0xc000185980, 0x6, 0x6, 0xc000201b80, 0xc000185980)
	/home/travis/gopath/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830 +0x29d
github.com/spf13/cobra.(*Command).ExecuteC(0xc000201900, 0x6, 0x10bbb98, 0x7)
	/home/travis/gopath/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/home/travis/gopath/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
github.com/golangci/golangci-lint/pkg/commands.(*Executor).Execute(...)
	/home/travis/gopath/src/github.com/golangci/golangci-lint/pkg/commands/executor.go:147
main.main()
	/home/travis/gopath/src/github.com/golangci/golangci-lint/cmd/golangci-lint/main.go:21 +0x7e
```

</details>

---

Using golangci-lint with versions <= v1.21.0 and the `unused` linter resulted in:
<details><summary>Output</summary>

```console
$ golangci-lint run --disable-all -E unused
var Build = "dev"
    ^
version.go:9:6: func `version` is unused (unused)
func version() string {
     ^
version.go:7:5: var `GitRev` is unused (unused)
var GitRev = "gitrevhere"
    ^
model/company.go:366:6: func `stringSliceToInterfaceSlice` is unused (unused)
func stringSliceToInterfaceSlice(l []string) []interface{} {
     ^
```
</details>

while using it with versions > v1.21.0:
<details><summary>Output</summary>

```console
$ golangci-lint run --disable-all -E unused
model/company.go:366:6: func `stringSliceToInterfaceSlice` is unused (unused)
version.go:9:6: func `version` is unused (unused)
version.go:7:5: var `GitRev` is unused (unused)
version.go:4:5: var `Build` is unused (unused)
```
</details>

As you can see, there are already differences in the 'default output'. When using it with code-climate  it results in the panic posted in `Verbose output of running`

**Comment by pedropombeiro** [comment]
Running into the same issue using it against the https://gitlab.com/gitlab-org/gitlab-runner source code. For now will need to stick with the v1.21.0 release, but ideally the regression would be fixed.

**Comment by l0nax** [comment]
We are currently facing the same problem.

Will there be a fix soon?

**Comment by ePirat** [comment]
Experiencing the same issue.

**Comment by l0nax** [comment]
Working fix: #1078

**Comment by ldez** [comment]
With the current version (v1.38.0), the problem seems fixed.

```console
$ golangci-lint run -v --disable-all -E unused --out-format code-climate
INFO [config_reader] Config search paths: [./ /home/ldez/sources/go/src/github.com/golangci/sandbox /home/ldez/sources/go/src/github.com/golangci /home/ldez/sources/go/src/github.com /home/ldez/sources/go/src /home/ldez/sources/go /home/ldez/sources /home/ldez /home /] 
INFO [lintersdb] Active 1 linters: [unused]       
INFO [loader] Go packages loading at mode 575 (deps|exports_file|files|imports|types_sizes|compiled_files|name) took 192.44582ms 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 268.24µs 
INFO [linters context/goanalysis] analyzers took 825.389477ms with top 10 stages: buildir: 824.971963ms, unused: 315.982µs, isgenerated: 77.555µs, directives: 23.977µs 
INFO [runner/max_same_issues] 1/4 issues with text "field `unexportedField` is unused" were hidden, use --max-same-issues 
INFO [runner] Issues before processing: 15, after processing: 14 
INFO [runner] Processors filtering stat (out/in): filename_unadjuster: 15/15, skip_files: 15/15, exclude: 15/15, diff: 15/15, max_per_file_from_linter: 15/15, max_same_issues: 14/15, severity-rules: 14/14, cgo: 15/15, skip_dirs: 15/15, autogenerated_exclude: 15/15, exclude-rules: 15/15, uniq_by_line: 15/15, path_shortener: 14/14, path_prefixer: 14/14, max_from_linter: 14/14, sort_results: 14/14, path_prettifier: 15/15, identifier_marker: 15/15, nolint: 15/15, source_code: 14/14 
INFO [runner] processing took 638.96µs with stages: exclude-rules: 279.974µs, nolint: 113.039µs, identifier_marker: 99.304µs, path_prettifier: 45.514µs, autogenerated_exclude: 29.696µs, source_code: 28.997µs, max_same_issues: 18.331µs, skip_dirs: 10.236µs, uniq_by_line: 3.457µs, path_shortener: 3.196µs, cgo: 2.175µs, max_from_linter: 1.591µs, filename_unadjuster: 1.375µs, max_per_file_from_linter: 850ns, diff: 265ns, skip_files: 242ns, severity-rules: 223ns, sort_results: 198ns, exclude: 182ns, path_prefixer: 115ns 
INFO [runner] linters took 1.810984168s with stages: unused: 1.810285848s 
[{"description":"unused: field `unexportedField` is unused","fingerprint":"CF3D8C59362510501CAD2FF2FFA78979","location":{"path":"sub/sub.go","lines":{"begin":19}}},{"description":"unused: field `unexportedField` is unused","fingerprint":"CF3D8C59362510501CAD2FF2FFA78979","location":{"path":"sub/sub.go","lines":{"begin":24}}},{"description":"unused: var `unexportedGlobalUnused` is unused","fingerprint":"F852AE104D1C9D888DFE9B4598386C03","location":{"path":"sub/sub.go","lines":{"begin":16}}},{"description":"unused: const `unexportedConstUnusedLocally` is unused","fingerprint":"DC765F5C3E48779953E0608B344595F0","location":{"path":"sub/sub.go","lines":{"begin":11}}},{"description":"unused: field `unexportedField` is unused","fingerprint":"01B42D111E15DF9A2C11D6FE08272624","location":{"path":"main.go","lines":{"begin":42}}},{"description":"unused: type `unusedStructWithMethods` is unused","fingerprint":"19E99BC666BCAF93F71CD5D7C5A225D8","location":{"path":"main.go","lines":{"begin":46}}},{"description":"unused: func `unusedStructWithMethods.unexportedMethod` is unused","fingerprint":"E4191BC2D5525F5166706C19947DEFD2","location":{"path":"main.go","lines":{"begin":51}}},{"description":"unused: func `usedStructWithMethods.unexportedMethod` is unused","fingerprint":"C523CFAE151B1980514C860E6F367830","location":{"path":"main.go","lines":{"begin":64}}},{"description":"unused: field `unusedField` is unused","fingerprint":"AB0BDFE517A4722EE4E9C0324B298B6E","location":{"path":"main.go","lines":{"begin":37}}},{"description":"unused: func `fn6` is unused","fingerprint":"203CBA67FF255A0E0D3ABB82E1479B73","location":{"path":"main.go","lines":{"begin":29}}},{"description":"unused: func `fn4` is unused","fingerprint":"0A52624CB1FB370E3B1C003121002405","location":{"path":"main.go","lines":{"begin":25}}},{"description":"unused: func `fn5` is unused","fingerprint":"7A3798F297235FD66CFDC1746F3F75B1","location":{"path":"main.go","lines":{"begin":27}}},{"description":"unused: func `unusedStructWithMethods.ExportedMethod` is unused","fingerprint":"E5B870D814DCA6B14784FF8D16372FF5","location":{"path":"main.go","lines":{"begin":55}}},{"description":"unused: type `unusedStruct` is unused","fingerprint":"07DA6447DC518DE7F8A65856E3177C48","location":{"path":"main.go","lines":{"begin":31}}}]INFO File cache stats: 2 entries of total size 1.5KiB 
INFO Memory: 22 samples, avg is 154.8MB, max is 206.7MB 
INFO Execution took 2.008836755s 
```
